src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -24,26 +24,29 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    UserService userService = UserServiceFactory.getUserService();
-
     String urlToRedirect = "/homePage";
     String userId = "null";
     String logoutUrl = "null";
     String loginUrl = "null";
     String userEmail = "null";
     Boolean showNewTab = false;
+    String error = null;
 
-    if (userService.isUserLoggedIn()) {
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()
+        && userService.getCurrentUser().getAuthDomain() == "google.com") {
       userId = userService.getCurrentUser().getUserId();
       userEmail = userService.getCurrentUser().getEmail();
       logoutUrl = userService.createLogoutURL(urlToRedirect);
       showNewTab = true;
+    } else if (userService.isUserLoggedIn()
+        && userService.getCurrentUser().getAuthDomain() != "google.com") {
+      error = "User need @google.com email address to log in";
```

Comment:
spelling: needs

Reply:
If this error is being sent down to the user, I would return just "Unauthorized".

Reply:
done